nomad/structs/structs.go

```diff
@@ -5184,6 +5184,31 @@ func (a *Allocation) RescheduleEligible(reschedulePolicy *ReschedulePolicy, fail
 	return attempted < attempts
 }
 
+// GCEligible returns if the allocation is eligible to be garbage collected
+// according to its terminal status and its reschedule trackers
+func (a *Allocation) GCEligible(reschedulePolicy *ReschedulePolicy, gcTime time.Time, thresholdIndex uint64) bool {
```

Comment:
I would prefer if this was just a helper function in the core_gc file or core_gc_helpers.go. It is super specific to that use case.